src/pudl/transform/classes.py

```diff
@@ -1230,15 +1234,32 @@ def rename_columns(
             "columns."
         )
 
-        # If we are attempting to rename columns that do *not* appear in the dataframe,
-        # warn the user.
-        missing_cols = set(params.columns).difference(set(df.columns))
-        if missing_cols:
-            logger.warning(
-                f"{self.table_id.value}: Attempting to rename columns which are not "
-                "present in the dataframe.\n"
-                f"Missing columns: {missing_cols}\nExisting Columns: {df.columns}"
+        # If we are actually renaming columns, do some defensive checks
+        if len(params.columns) > 0:
+            # If we are attempting to rename columns that do *not* appear in the dataframe,
+            # raise an error.
+            missing_cols = set(params.columns).difference(set(df.columns))
+            if missing_cols:
+                raise ValueError(
+                    f"{self.table_id.value}: Attempting to rename columns which are not "
+                    "present in the dataframe.\n"
+                    f"Missing columns: {missing_cols}\nExisting Columns: {df.columns}"
+                )
+            # If we fail to rename any columns that do appear in the dataframe,
+            # other than those we expect (not_renamed_columns), raise an error.
+            logger.warn(f"Ignore cols: {params.not_renamed_columns}")
+            unmapped_cols = (
+                set(df.columns)
+                .difference(set(params.columns))
+                .difference(set(params.not_renamed_columns))
             )
+            logger.warn(f"unmapped_cols is {unmapped_cols}")
+            if unmapped_cols:
+                raise ValueError(
+                    f"{self.table_id.value}: The following columns are present in the"
+                    " in the dataframe but not renamed.\n"
+                    f"Unmapped columns: {unmapped_cols}\nExisting Columns: {df.columns}"
+                )
```

Comment:
Does this check necessitate a bunch of new transformation metadata? If so it'd be great to catch both extra and missing columns. I noticed that we check for both kinds of errors in the EIA data but only one kind of error here, but it seemed like it might be a hassle to compile and verify the necessary metadata.

```suggestion
            if unmapped_cols:
                raise ValueError(
                    f"{self.table_id.value}: The following columns are present in the"
                    " dataframe but are not renamed.\n"
                    f"Unmapped columns: {unmapped_cols}\nExisting Columns: {df.columns}"
                )
```